ruby/red-arrow/ext/arrow/raw-records.cpp

```diff
@@ -26,7 +26,9 @@ namespace red_arrow {
       explicit RawRecordsBuilder(VALUE records, int n_columns)
         : Converter(),
           records_(records),
-          n_columns_(n_columns) {
+          record_(Qnil),
+          n_columns_(n_columns),
+          is_produce_mode_(false) {
```

Comment:
How about creating `RawRecordProducer` instead of adding a new "produce" mode to `RawRecordsBuilder`?

Reply:
fix: [3318abc](https://github.com/apache/arrow/pull/37137/commits/3318abc8aa4e7d9e82da61a6b779d7556d97bee6)
Sounds pretty nice to me!
I tried to split `RawRecordProducer` into its own source file because some methods had to know the row-wise processing situation. 
What do you think of it?

Reply:
Ah, I should have written `RawRecordsProducer` not `RawRecordProducer` because it produces multiple raw records.

So I think that we can still use `raw-records.cpp` instead of `raw-record.cpp` because it also handles multiple raw records not just one raw record.

Reply:
fix: [d3e5f3f](https://github.com/apache/arrow/pull/37137/commits/d3e5f3f0307bb7036a02e32d26c17f8d7c68af88)